feat(Skeleton): support `classNames` and `styles`

coding-ice opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is confirmed.
Your pull requests will be merged after one of the collaborators approve.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature

### 🔗 Related Issues
https://github.com/ant-design/ant-design/issues/52115


| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Enhance Skeleton components with classNames and styles support           |
| 🇨🇳 Chinese | 增加语义 classNames 和 styles 属性          |

coding-ice committed:
feat: Add Skeleton component configuration support in ConfigProvider

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
feat: Enhance ConfigProvider with Skeleton component support

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
feat: Enhance Skeleton components with classNames and styles support

coding-ice committed:
feat: Update Skeleton documentation with common API and semantic structure enhancements

coding-ice committed:
test: Enhance Skeleton component tests with custom styles and semantic element support

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
chore: update snapshots for Skeleton component tests to reflect changes in expected output

thinkasany commented on components/skeleton/demo/_semantic.tsx:
```diff
@@ -0,0 +1,44 @@
+  en: {
+    root: 'Root element',
+    header: 'Header element',
+    content: 'Content element',
```
这个我仅是记录一下，和豆酱老师确认一下，先不需要修改。因为看起来是包裹了title和其他的内容，不确定用 section会不会更合适一点。
 cc @zombieJ 
 
<img width="452" alt="image" src="https://github.com/user-attachments/assets/9dcb0fff-6c70-4765-affe-944b31c94619" />

<img width="555" alt="image" src="https://github.com/user-attachments/assets/8ef49127-7dc8-412d-9c36-c61cb2acd237" />

zombieJ replied:
content 改 section，其他的我 OK

thinkasany replied:
@coding-ice content 换成 section吧，v6的语义化中 content目前不会当 parent node, 这边用section 是更合适的~。

thinkasany commented on components/skeleton/demo/_semantic.tsx:
```diff
@@ -0,0 +1,44 @@
+  },
+  en: {
+    root: 'Root element',
+    header: 'Header element',
```
其他地方header都是拿来包裹title的，这里只包裹了个avatar，感觉似乎也需要考虑是否找个更合适的语义。
<img width="432" alt="image" src="https://github.com/user-attachments/assets/e63196c8-2d1b-4a64-b362-a2fdb91c69bd" />

thinkasany replied:
@coding-ice 有没有更好的建议呢，这边的header包裹avatar没有前车之鉴，是唯一一处这么使用的。暂时没有合适的参考命名，就是可以先这么用着，但是如果有更好的命名的话，可以大家一起讨论一下。
![a7d436c41dc2d87eed6dc11f78c5d46c](https://github.com/user-attachments/assets/91f771d9-945a-47d5-ae62-1d23ac969f7b)

coding-ice replied:
这个如何：ant-skeleton-figure
https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/figure

coding-ice replied:
这个先过好了，到时候有更好的命令再改也行

thinkasany commented on components/skeleton/demo/_semantic_node.tsx:
```diff
@@ -0,0 +1,32 @@
+const locales = {
+  cn: {
+    root: '根元素',
+    node: '节点元素 (可以是 avatar、button、input)',
```
是不是也应该加个button和input的，看起来上面的demo比这个文字更清晰
<img width="131" alt="image" src="https://github.com/user-attachments/assets/de7107e6-4723-4ead-9052-8af0346c7cf0" />

<img width="850" alt="image" src="https://github.com/user-attachments/assets/534f62d6-76fa-4b8a-b330-f9a5259e21be" />

coding-ice replied:
只是一个内容的 cls 不一样 如果都列出来 会不会显得太多了，不过可以考虑放到一个 demo 里面

thinkasany replied:
对，就是可以考虑一下 是不是需要 “放到一个 demo”，来更好的展示～

thinkasany commented on components/skeleton/Skeleton.tsx:
```diff
@@ -18,6 +18,8 @@ import Title from './Title';
 /* This only for skeleton internal. */
 type SkeletonAvatarProps = Omit<AvatarProps, 'active'>;
 
+export type SemanticName = 'root' | 'header' | 'content' | 'avatar' | 'title' | 'paragraph';
```
是不是还有input
<img width="570" alt="image" src="https://github.com/user-attachments/assets/17cca336-8bb7-44c0-b8d4-f36fdad8bd20" />

coding-ice committed:
test: Refactor Skeleton component tests to apply distinct custom styles for each semantic element

coding-ice committed:
refactor: Update Skeleton components to replace 'with-element' class with 'element' class for consistency

coding-ice committed:
chore: Update snapshots for Skeleton component tests to reflect class name changes from 'with-element' to 'element'

thinkasany commented on components/skeleton/Button.tsx:
```diff
@@ -17,31 +17,41 @@ const SkeletonButton: React.FC<SkeletonButtonProps> = (props) => {
+    style,
+    styles,
     block = false,
     size = 'default',
   } = props;
```
```suggestion
    size = 'default',
    ...restProps
  } = props;
```

thinkasany commented on components/skeleton/Button.tsx:
```diff
@@ -17,31 +17,41 @@ const SkeletonButton: React.FC<SkeletonButtonProps> = (props) => {
   const [wrapCSSVar, hashId, cssVarCls] = useStyle(prefixCls);
 
-  const otherProps = omit(props, ['prefixCls']);
+  const otherProps = omit(props, ['prefixCls', 'className', 'classNames', 'style', 'styles']);
```
```suggestion
```

thinkasany replied:
是不是可以换成解构剩余参数

thinkasany commented on components/skeleton/Button.tsx:
```diff
@@ -17,31 +17,41 @@ const SkeletonButton: React.FC<SkeletonButtonProps> = (props) => {
+        className={skeletonButtonClassNames?.button}
+        style={{ ...styles?.button, ...style }}
+        size={size}
+        {...otherProps}
```
```suggestion
        {...restProps}
```

coding-ice committed:
refactor: Replace 'omit' utility with spread operator for props in Skeleton components and update class name from 'with-element' to 'element'

coding-ice committed:
refactor: Rename Skeleton.Node to Skeleton.Element and update related demo files

coding-ice committed:
refactor: Rename LocaleMap to LOCALE_MAP for consistency in semantic element handling

thinkasany commented on components/skeleton/demo/_semantic_element.tsx:
```diff
@@ -0,0 +1,86 @@
+
+  return (
+    <Flex vertical style={{ width: '100%' }}>
+      <Radio.Group value={element} onChange={(e) => setElement(e.target.value)}>
```
推荐使用 options 的写法 或换成 segmented
<img width="777" alt="image" src="https://github.com/user-attachments/assets/5c3289a5-8b3a-42c2-b162-36ea178b839f" />

coding-ice committed:
refactor: Simplify Radio.Group in PreviewContent by using OPTIONS array for element selection

coding-ice committed:
refactor: Replace Radio.Group with Segmented for element selection in PreviewContent

coding-ice committed:
refactor: Update Skeleton component to replace 'content' with 'section' semantic element and adjust related styles and tests

coding-ice committed:
refactor: Update Skeleton component snapshots to replace 'ant-skeleton-content' with 'ant-skeleton-section' for improved semantic clarity

coding-ice committed:
test: snap

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
Merge branch 'next' into refactor/skeleton

thinkasany commented:
冲突了，空的时候处理一下，方便豆酱老师review

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice commented:
@afc163  佬，这个需不需要考虑加一个排序的 commit 的钩子啊，这个冲突时排序导致的

coding-ice committed:
Merge branch 'refactor/skeleton' of ice:GetWebHB/ant-design into refactor/skeleton

coding-ice commented on components/skeleton/demo/_semantic_element.tsx:
```diff
@@ -0,0 +1,104 @@
+
+const OPTIONS = [
+  {
+    value: 'avatar',
```
其实一个 `Node` 感觉都够了，用户自己自定义就好，像这种一般都是自己自定义的吧

thinkasany replied:
还少了 Image,  我觉得还好，因为都 一 一 对应了 antd的组件，对于用户来说可以开箱即用，实际上还是有应用场景的。
![image](https://github.com/user-attachments/assets/f324e17c-1046-4df5-aff4-d991c1bf2156)

coding-ice replied:
`Image` 是 `Node` 封装的，所以没有列出来，要不文档中我来注明一下

thinkasany replied:
先看看豆酱老师怎么说吧。

zombieJ replied:
我意思是不要具体到 `classNames.button` 这种，因为 Button 组件的语义化是没有 `classNames.button`。所以按理说都应该叫 `classNames.content`

coding-ice committed:
refactor(skeleton): Rename semantic elements from specific names to generic 'content'

This change updates the Skeleton components to use a more generic 'content' semantic element instead of specific names like 'avatar', 'button', etc. The modifications include:
- Updating Element.tsx to use a simplified semantic type
- Modifying class and style references in Avatar, Button, Input, and Node components
- Updating test cases and demo to reflect the new semantic naming

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
Merge branch 'next' into refactor/skeleton

thinkasany commented on components/skeleton/Skeleton.tsx:
```diff
@@ -119,13 +129,22 @@ const Skeleton: React.FC<SkeletonProps> & CompoundedComponent = (props) => {
     let avatarNode: React.ReactNode;
     if (hasAvatar) {
       const avatarProps: SkeletonAvatarProps = {
+        className: classNames(contextClassNames?.avatar, skeletonClassNames?.avatar),
```
```suggestion
        className: classNames(contextClassNames.avatar, skeletonClassNames?.avatar),
```

thinkasany commented on components/skeleton/Button.tsx:
```diff
@@ -17,31 +16,41 @@ const SkeletonButton: React.FC<SkeletonButtonProps> = (props) => {
   const prefixCls = getPrefixCls('skeleton', customizePrefixCls);
   const [wrapCSSVar, hashId, cssVarCls] = useStyle(prefixCls);
 
-  const otherProps = omit(props, ['prefixCls']);
```
你说这个无用的omit的剔除要不要再master改一下，然后慢慢的落下来。 毕竟合并这个pr也没这么快，真到发布上去v6就更久了，master上的代码留着这么没用的代码，怪怪的。

coding-ice replied:
应该是不行的，如果删了原本应该被传递下去的，反而没传递了

比如：className

```jsx
const otherProps = omit(props, ['prefixCls']);
<Element {...otherProps} />;
```

coding-ice committed:
Update components/skeleton/Skeleton.tsx

Co-authored-by: thinkasany <480968828@qq.com>
Signed-off-by: ice <49827327+coding-ice@users.noreply.github.com>

thinkasany commented on components/skeleton/Skeleton.tsx:
```diff
@@ -173,6 +203,9 @@ const Skeleton: React.FC<SkeletonProps> & CompoundedComponent = (props) => {
         [`${prefixCls}-round`]: round,
       },
+      contextClassNames?.root,
+      // skeleton?.className,
```
为啥这里是注释了的

thinkasany commented on components/skeleton/demo/_semantic_element.tsx:
```diff
@@ -0,0 +1,88 @@
+
+  return (
+    <Flex vertical style={{ width: '100%' }}>
+      <Segmented
```
这个宽度调整一下 会不会好一点
![image](https://github.com/user-attachments/assets/5cc487b8-6ec2-4553-ae8d-2e5ea2285f95)

感觉现在这样子的不太好看
![image](https://github.com/user-attachments/assets/9498cfed-1886-4522-b2ed-1aabe19ca315)

coding-ice committed:
refactor(Skeleton): Remove commented-out className assignment

coding-ice committed:
sty;e(Skeleton): Adjust demo component layout with fit-content width

coding-ice committed:
Merge branch 'next' into refactor/skeleton

thinkasany committed:
Merge branch 'next' into refactor/skeleton

zombieJ commented on components/skeleton/Image.tsx:
```diff
@@ -1,45 +1,28 @@
-      </div>
-    </div>,
+  return (
+    <SkeletonNode {...props}>
```
这个 className 不对，少了 `ant-skeleton-image`，变成 `ant-skeleton-node` 了：
<img width="365" alt="截屏2025-02-08 11 06 33" src="https://github.com/user-attachments/assets/8032bbb0-d41d-4045-8327-2a43b44fa94d" />

coding-ice replied:
这个我有空处理下 因为原来的 `Node` 也叫 image，想着不如改为 `Node` 稍微合理些
还是独立出来好了

thinkasany replied:
独立吧，不然设置node就联动影响了

coding-ice committed:
Merge branch 'next' into refactor/skeleton

coding-ice committed:
feat: Enhance Skeleton Image component with improved styling and customization

thinkasany committed:
Merge branch 'next' into refactor/skeleton
